src/histogram.rs

```diff
@@ -0,0 +1,118 @@
+/// Structure used to create a distribution of node
+/// centrality values for use in a displayed histogram
+
+#[derive(Default, Clone)]
+pub struct Histogram {
+    /// Minimum value of a factor.
+    pub min: f64,
+    /// Maximum value of a factor.
+    pub max: f64,
+    /// store values
+    pub values: Vec<f64>,
+}
+
+impl Histogram {
+    pub fn default() -> Self {
+        Self {
+            min: f64::MAX,
+            max: f64::MIN,
+            values: Vec::new(),
+        }
+    }
+
+    // Store a single value
+    pub fn add(&mut self, value: f64) {
+        self.values.push(value);
+    }
+
+    // Compute the histogram tables, returning that table along
+    // with the value of the maximum count in a single slot.
+    pub fn compute(&mut self, num_counts: usize) -> (Vec<usize>, usize) {
+        for value in self.values.iter() {
+            if *value < self.min {
+                self.min = *value;
+            }
+            if *value > self.max {
+                self.max = *value;
+            }
+        }
+
+        let delta = self.max - self.min;
+        let mut counts = vec![0; num_counts];
+        let mut max_count: usize = 0;
+
+        if delta == 0.0 {
+            return (counts, max_count);
+        }
+
+        for value in self.values.iter() {
+            let part = (*value - self.min) / delta;
+            let mut slot = (part * num_counts as f64).floor() as usize;
+            if slot >= num_counts {
+                slot = num_counts - 1;
+            }
+            counts[slot] += 1;
```

Comment:
Probably never happen as we'd not have such long histograms, just to know that there could be construction like:
`counts[slot].saturating_add(1);`
to prevent overflow and counting from zero 😃